qiskit/algorithms/algorithm_result.py

```diff
@@ -14,37 +14,25 @@
 This module implements the abstract base class for algorithm results.
 """
 
-from typing import Optional, Dict, Tuple
 from abc import ABC
-import collections
 import inspect
+from collections import OrderedDict
+import pprint
 
 
-class AlgorithmResult(ABC, collections.UserDict):
+class AlgorithmResult(ABC):
     """ Abstract Base Class for algorithm results."""
 
-    def __init__(self, a_dict: Optional[Dict] = None) -> None:
-        super().__init__()
-        if a_dict:
-            self.data.update(a_dict)
-
-    def __setitem__(self, key: object, item: object) -> None:
-        raise TypeError("'__setitem__' invalid for this object.")
-
-    def __delitem__(self, key: object) -> None:
-        raise TypeError("'__delitem__' invalid for this object.")
-
-    def clear(self) -> None:
-        raise TypeError("'clear' invalid for this object.")
-
-    def pop(self, key: object, default: Optional[object] = None) -> object:
-        raise TypeError("'pop' invalid for this object.")
+    def __repr__(self) -> str:
```

Comment:
Should this be `__str__` rather than repr since repr is supposed to be able to be eval'd into an instance which is no longer the case given this is not a dict.

Reply:
Changed to __str__